### ISSUE TITLE
realm: try to account for static TLS when specifying thread stack sizes

### ISSUE THREAD

**Issue by streichler** [StanfordLegion/legion#843]
Realm uses a number of threads with very small stacks (e.g. the kernel thread that hosts user threads), and the creation of those threads can fail when an app uses a large amount of TLS.  We need a way to detect the amount of stack space that is needed for TLS or at least a way for an application to specify it.

**Comment by lightsighter** [comment]
Can you give a definition of what "a large amount of TLS" is? I realize it may be OS specific.

**Comment by streichler** [comment]
> Can you give a definition of what "a large amount of TLS" is? I realize it may be OS specific.

The following is sufficient to make Realm fail on initialization on a Ubuntu 16.04 system.  It's likely that slightly smaller values would cause nastier problems as the threads are created successfully but then fall off the bottom of their stacks during execution.
```
struct LargeStruct {
  char data[256000];
};

__thread LargeStruct large_tls_var;
```

**Comment by streichler** [comment]
This was fixed in master with commit 8df57f6.
